app-frontend/src/app/core/services/dagUtils.service.js

```diff
@@ -0,0 +1,88 @@
+export default (app) => {
+    class DagUtils {
```

Comment:
Slight quibble with this name: since we already have DAGs in the context of Airflow, perhaps this could be `ToolDagUtils` or `LabDagUtils` or something a bit less general.